src/Domain/CoordinateMaps/CoordinateMap.tpp

```diff
@@ -351,6 +351,142 @@ auto CoordinateMap<SourceFrame, TargetFrame, Maps...>::jacobian_impl(
   return jac;
 }
 
+namespace detail {
+template <typename T, typename Map, size_t Dim,
+          Requires<not domain::is_map_time_dependent_v<Map>> = nullptr>
+std::array<T, Dim> get_frame_velocity(
+    const Map& /*the_map*/, const std::array<T, Dim>& /*point*/,
+    const double /*t*/,
+    const std::unordered_map<
+        std::string, std::unique_ptr<domain::FunctionsOfTime::FunctionOfTime>>&
+    /*funcs_of_time*/) noexcept {
+  return std::array<T, Dim>{};
+}
+
+template <typename T, typename Map, size_t Dim,
+          Requires<domain::is_map_time_dependent_v<Map>> = nullptr>
+std::array<T, Dim> get_frame_velocity(
+    const Map& the_map, const std::array<T, Dim>& point, const double t,
+    const std::unordered_map<
+        std::string, std::unique_ptr<domain::FunctionsOfTime::FunctionOfTime>>&
+        funcs_of_time) noexcept {
+  return the_map.frame_velocity(point, t, funcs_of_time);
+}
+}  // namespace detail
+
+template <typename SourceFrame, typename TargetFrame, typename... Maps>
+template <typename T>
+auto CoordinateMap<SourceFrame, TargetFrame, Maps...>::
+    coords_frame_velocity_jacobians_impl(
+        tnsr::I<T, dim, SourceFrame> source_point, const double time,
+        const std::unordered_map<
+            std::string,
+            std::unique_ptr<domain::FunctionsOfTime::FunctionOfTime>>&
+            functions_of_time) const noexcept
+    -> std::tuple<tnsr::I<T, dim, TargetFrame>,
+                  InverseJacobian<T, dim, SourceFrame, TargetFrame>,
+                  Jacobian<T, dim, SourceFrame, TargetFrame>,
+                  tnsr::I<T, dim, TargetFrame>> {
+  std::array<T, dim> mapped_point = make_array<T, dim>(std::move(source_point));
+  InverseJacobian<T, dim, SourceFrame, TargetFrame> inv_jac{};
+  Jacobian<T, dim, SourceFrame, TargetFrame> jac{};
+  tnsr::I<T, dim, TargetFrame> frame_velocity{};
+
+  tuple_transform(
+      maps_,
+      [&frame_velocity, &inv_jac, &jac, &mapped_point, time,
+       &functions_of_time](const auto& map, auto index,
+                           const std::tuple<Maps...>& maps) noexcept {
+        constexpr const size_t count = decltype(index)::value;
+        using Map = std::decay_t<decltype(map)>;
+
+        tnsr::Ij<T, dim, Frame::NoFrame> noframe_jac{};
```

Comment:
Why not declare these outside the transform and capture?

Reply:
I prefer tighter scoping of variables because it makes code easier to reason about. I capture the others because they're "returned by reference"